llvm/cmake/modules/HandleLLVMOptions.cmake

```diff
@@ -289,6 +289,39 @@ function(has_msvc_incremental_no_flag flags incr_no_flag_on)
   endif()
 endfunction()
 
+macro(add_compile_option_ext flag name)
+  cmake_parse_arguments(ARG "" "" "" ${ARGN}) 
+  set(CHECK_STRING "${flag}")
+  if (MSVC)
+    set(CHECK_STRING "/WX ${CHECK_STRING}")
+  else()
+    set(CHECK_STRING "-Werror ${CHECK_STRING}")
+  endif()
```

Comment:
Why do we need this? `check_*_compiler_flag` are supposed to check that the flag is supported without a diagnostic already.